--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -10,7 +10,7 @@
 set(CMAKE_AUTOMOC ON)
 set(CMAKE_INCLUDE_CURRENT_DIR ON) # This is to find generated *.moc and *.h files in build dir
 
-add_library(SortFilterProxyModel OBJECT
+add_library(SortFilterProxyModel
     qqmlsortfilterproxymodel.cpp
     filters/filter.cpp
     filters/filtercontainer.cpp
@@ -47,3 +47,12 @@
     $<TARGET_PROPERTY:Qt5::Core,INTERFACE_INCLUDE_DIRECTORIES>
     $<TARGET_PROPERTY:Qt5::Qml,INTERFACE_INCLUDE_DIRECTORIES>
     )
+
+install(TARGETS SortFilterProxyModel
+    ARCHIVE DESTINATION lib
+    PUBLIC_HEADER DESTINATION include
+)
+install(FILES qqmlsortfilterproxymodel.h DESTINATION include)
+install(FILES filters/filtercontainer.h DESTINATION include/filters)
+install(FILES sorters/sortercontainer.h DESTINATION include/sorters)
+install(FILES proxyroles/proxyrolecontainer.h DESTINATION include/proxyroles)
